lambdas/data-migration2/tests/test-granulesAndFiles.js

```diff
@@ -591,6 +599,90 @@ test.serial('migrateGranuleRecord updates an already migrated record if the upda
   t.deepEqual(record.updated_at, new Date(newerGranule.updatedAt));
 });
 
+test.serial('migrateGranuleRecord supports undefined values in dynamo and overwrites defined values in Postgres when re-migrating', async (t) => {
+  const {
+    knex,
+    granulePgModel,
+    testCollection,
+    testExecution,
+    testProvider,
+    testPdr,
+  } = t.context;
+
+  const testGranule = generateTestGranule({
+    collectionId: constructCollectionId(testCollection.name, testCollection.version),
+    queryFields: { foo: cryptoRandomString({ length: 8 }) },
+    execution: testExecution.url,
+    provider: testProvider.name,
+    pdrName: testPdr.name,
+    status: 'completed',
+    version: cryptoRandomString({ length: 3 }),
+    updatedAt: Date.now() - 1000,
+  });
+
+  await createRejectableTransaction(knex, (trx) => migrateGranuleRecord(testGranule, trx));
+
+  // get a list of nullable granule keys
+  const nonNullablefields = [
+    'granuleId',
+    'collectionId',
+    'createdAt',
+    'updatedAt',
+    'status',
+    'execution',
+  ];
+  const nullableGranuleFields = omit(testGranule, nonNullablefields);
+
+  // Create object with only nullable fields as { field: undefined }
+  const undefinedGranuleFields = {};
+  Object.keys(nullableGranuleFields).forEach((field) => {
+    undefinedGranuleFields[field] = undefined;
+  });
+
+  const newerGranule = {
+    ...testGranule,
+    ...undefinedGranuleFields,
+    updatedAt: Date.now(),
+  };
+
+  const granuleCumulusId = await createRejectableTransaction(
+    knex,
+    (trx) => migrateGranuleRecord(newerGranule, trx)
+  );
+  const record = await granulePgModel.get(knex, {
+    cumulus_id: granuleCumulusId,
+  });
+
+  // Convert the granule payload we're migrating to the Postgres Granule format
+  const expectedPgGranule = await translateApiGranuleToPostgresGranule({
+    dynamoRecord: newerGranule,
+    knexOrTransaction: knex,
+  });
+
+  t.teardown(async () => {
+    await t.context.granulePgModel.delete(t.context.knex, { cumulus_id: record.cumulus_id });
+  });
+
+  // The expectedPgGranule is an object created outside of PG and does not have foreign keys
+  // Remove null properties from the PG record before comparison. The second migration should have
+  // replaced valid values with null values.
+  const omitList = ['cumulus_id', 'pdr_cumulus_id', 'provider_cumulus_id'];
+  t.deepEqual(omit(removeNilProperties(record), omitList), expectedPgGranule);
```

Comment:
Nit: might it be clearer to add the missing values to the expected granule?  e.g.: 

```
t.deepEqual(removeNilProperties(record)), { ...expectedPgGranule, cumulus_id: record.cumulus_id, <etc> });
```

Reply:
Yup! So we talked about this and I've removed `pdr_cumulus_id` and `provider_cumulus_id` from the list of fields we're excluding. We expect those to both be `null` if we're migrating a record with a `null` `pdrName` and `provider`.

https://github.com/nasa/cumulus/pull/3082/commits/d0b06777302876acea8697937c988717ac2f9b7f